uaclient/event_logger.py

```diff
@@ -145,6 +178,56 @@ def _generate_failed_services(self):
         }
         return list(set.union(self._failed_services, services_with_error))
 
+    def _process_events_services(self):
+        response = {
+            "_schema_version": JSON_SCHEMA_VERSION,
+            "result": "success" if not self._error_events else "failure",
+            "processed_services": sorted(self._processed_services),
+            "failed_services": sorted(self._generate_failed_services()),
+            "errors": self._error_events,
+            "warnings": self._warning_events,
+            "needs_reboot": self._needs_reboot,
+        }
+
+        print(json.dumps(response, sort_keys=True))
+
+    def _format_status_output(self, status: Dict[str, Any]) -> Dict[str, Any]:
+        status["environment_vars"] = [
+            {"name": name, "value": value}
+            for name, value in sorted(os.environ.items())
+            if name.lower() in CONFIG_FIELD_ENVVAR_ALLOWLIST
+            or name.startswith("UA_FEATURES")
+            or name == "UA_CONFIG_FILE"
+        ]
+
+        if not status.get("simulated"):
+            available_services = [
+                service
+                for service in status.get("services", [])
+                if service.get("available", "yes") == "yes"
+            ]
+            status["services"] = available_services
+
+        # We don't need the origin info in the json output
+        status.pop("origin", "")
```

Comment:
This seems like it belongs over in the `action.status` function

Reply:
Missed this after the rebase - will change

Reply:
Ended up moving it back to `status.py` - the CLI formatting function is also there, and given it is called conditionally based on the event logger mode, it would bloat the action